### PR TITLE
Fix various issues with tests

### PR DESCRIPTION
**Pull request by emerkle826**
- Fixes  OSGi tests since Awaitility was added
- Fixes GraphPagingIT tests against DSE 6.8
- Increases timeout in MetadataManagerTests to try to make the tests less flaky

### CODE REVIEW

File: core/src/main/java/com/datastax/dse/driver/internal/core/graph/ContinuousGraphRequestHandler.java [hunk 1]
```diff
@@ -99,7 +99,7 @@ protected Duration getGlobalTimeout() {
   @NonNull
   @Override
   protected Duration getPageTimeout(int pageNumber) {
-    return Duration.ZERO;
+    return this.globalTimeout;
```
Comment (emerkle826) [inline core/src/main/java/com/datastax/dse/driver/internal/core/graph/ContinuousGraphRequestHandler.java]: I have concerns about the full effect of this change. Currently, a value of 0 means that the driver will wait forever for a response to successive page requests (the first page timeout is controlled by the globalTimeout).

I changed the return value here to be the globalTimeout value as the handler doesn't have a separate page timeouts (like [ContinuousCqlRequestHandler](https://github.com/datastax/java-driver/blob/4.x/core/src/main/java/com/datastax/dse/driver/internal/core/cql/continuous/ContinuousCqlRequestHandler.java#L51-L52)).

The test that fails is [here](https://github.com/datastax/java-driver/pull/1431/files#diff-c96ea9340f349d35ca1642eb4f50c76aR490), as it expects the second page request to timeout while CCM has been paused. But with this method returning a Duration of 0, it never times out the request. Instead, we get a Heartbeat timeout 30 seconds later. This also generally causes many of the DatatProvider tests in that class to fail as more queries are attempted before the session has been fully re-established after CMM is resumed, which causes the session to die with an AllNodesFailed exception.

I'm not sure if this is the best solution, as maybe GraphPaging needs to support setting a timeout for `otherPages` like the ContiuousCqlRequestHandler does now.

File: integration-tests/src/test/java/com/datastax/dse/driver/api/core/graph/GraphPagingIT.java [hunk 2]
```diff
@@ -69,6 +69,9 @@
                   .withStringList(
                       DefaultDriverOption.METRICS_NODE_ENABLED,
                       Collections.singletonList(DseNodeMetric.GRAPH_MESSAGES.getPath()))
+                  // Increase heartbeat interval to avoid interferences since we pause/stop the
+                  // server in this test
+                  .withDuration(DefaultDriverOption.HEARTBEAT_INTERVAL, Duration.ofMinutes(10))
```
Comment (emerkle826) [inline integration-tests/src/test/java/com/datastax/dse/driver/api/core/graph/GraphPagingIT.java]: If we remove the invalid test below, I don't think we need to increase the heartbeat here. The rest of the tests in this class pass fine without even coming close to tripping the 30 second heartbeat timeout.

File: integration-tests/src/test/java/com/datastax/dse/driver/api/core/graph/GraphPagingIT.java [hunk 3]
```diff
@@ -461,38 +464,6 @@ public void should_trigger_global_timeout_async() throws InterruptedException {
     }
   }
 
-  @Test
-  public void should_trigger_global_timeout_async_after_first_page() throws InterruptedException {
```
Comment (tomekl007) [inline integration-tests/src/test/java/com/datastax/dse/driver/api/core/graph/GraphPagingIT.java:465]: Why do you want to remove this test? It exercises the global timeout for async paging.

Comment (adutra) [reply integration-tests/src/test/java/com/datastax/dse/driver/api/core/graph/GraphPagingIT.java:465]: Discussed in Slack: TLDR is this test is too broken to be kept.

File: core/src/test/java/com/datastax/oss/driver/internal/core/metadata/MetadataManagerTest.java [hunk 4]
```diff
@@ -308,7 +308,7 @@ private void waitForPendingAdminTasks() {
     // This works because the event loop group is single-threaded
     Future<?> f = adminEventLoopGroup.schedule(() -> null, 5, TimeUnit.NANOSECONDS);
     try {
-      Uninterruptibles.getUninterruptibly(f, 100, TimeUnit.MILLISECONDS);
+      Uninterruptibles.getUninterruptibly(f, 500, TimeUnit.MILLISECONDS);
```
Comment (adutra) [inline core/src/test/java/com/datastax/oss/driver/internal/core/metadata/MetadataManagerTest.java]: Are we waiting for some specific event to occur? If so, would it be possible to use Awaitility to wait on that condition?

File: integration-tests/src/test/java/com/datastax/oss/driver/osgi/support/BundleOptions.java [hunk 5]
```diff
@@ -167,6 +168,14 @@ public static CompositeOption simulacronBundles() {
                 simulacronVersion));
   }
 
+  public static CompositeOption awaitilityBundles() {
```
Comment (adutra) [inline integration-tests/src/test/java/com/datastax/oss/driver/osgi/support/BundleOptions.java:171]: Note to myself: I need to port this to JAVA-2658.

File: core/src/test/java/com/datastax/oss/driver/internal/core/metadata/MetadataManagerTest.java (hunk 4 above)
Comment (emerkle826) [reply core/src/test/java/com/datastax/oss/driver/internal/core/metadata/MetadataManagerTest.java]: I'm not sure,but I'll investigate further. With the timeout set to 100, I couldn't reproduce the failure locally, and it failed sporadically on Jenkins/CloudBees.
I can lower the value and reproduce it and figure out what I need to be waiting on.

File: integration-tests/src/test/java/com/datastax/oss/driver/osgi/support/BundleOptions.java (hunk 5 above)
Comment (adutra) [reply integration-tests/src/test/java/com/datastax/oss/driver/osgi/support/BundleOptions.java:171]: Done.

File: core/src/test/java/com/datastax/oss/driver/internal/core/metadata/MetadataManagerTest.java [hunk 6]
```diff
@@ -290,6 +287,8 @@ public void should_remove_node() {
   private static class TestMetadataManager extends MetadataManager {
 
     private List<MetadataRefresh> refreshes = new CopyOnWriteArrayList<>();
+    private int addNodeCount = 0;
```
Comment (adutra) [inline core/src/test/java/com/datastax/oss/driver/internal/core/metadata/MetadataManagerTest.java]: These 2 fields probably should be volatile.

File: core/src/test/java/com/datastax/oss/driver/internal/core/metadata/MetadataManagerTest.java [hunk 7]
```diff
@@ -301,18 +300,24 @@ Void apply(MetadataRefresh refresh) {
...
-    Future<?> f = adminEventLoopGroup.schedule(() -> null, 5, TimeUnit.NANOSECONDS);
-    try {
-      Uninterruptibles.getUninterruptibly(f, 100, TimeUnit.MILLISECONDS);
-    } catch (ExecutionException e) {
-      fail("unexpected error", e.getCause());
-    } catch (TimeoutException e) {
-      fail("timed out while waiting for admin tasks to complete", e);
-    }
+  private void waitForPendingAdminTasks(Callable<Boolean> condition) {
+    await().atMost(500, TimeUnit.MILLISECONDS).until(condition);
```
Comment (adutra) [inline core/src/test/java/com/datastax/oss/driver/internal/core/metadata/MetadataManagerTest.java:325]: Nice rework! 👍

**Review by adutra — CHANGES_REQUESTED** [review]

File: core/src/test/java/com/datastax/oss/driver/internal/core/metadata/MetadataManagerTest.java (hunk 6 above)
Comment (emerkle826) [reply core/src/test/java/com/datastax/oss/driver/internal/core/metadata/MetadataManagerTest.java]: Good catch. I did try to verify that the tests run in parallel don't trample each other here, and it seemed like they didn't, but `volatile` here makes sense.

**Review by adutra — APPROVED** [review]

**Pull request merged**